cmd/crank/beta/validate/loader_test.go

```diff
@@ -0,0 +1,229 @@
+/*
+Copyright 2024 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validate
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var (
+	coolResource = map[string]interface{}{
+		"apiVersion": "example.org/v1alpha1",
+		"kind":       "ComposedResource",
+		"metadata": map[string]interface{}{
+			"annotations": map[string]interface{}{
+				"crossplane.io/composition-resource-name": "resource-a",
+			},
+			"name": "test-render-a",
+		},
+		"spec": map[string]interface{}{
+			"coolField": "I'm cool!",
+		},
+	}
+	coolerResource = map[string]interface{}{
+		"apiVersion": "example.org/v1alpha1",
+		"kind":       "ComposedResource",
+		"metadata": map[string]interface{}{
+			"annotations": map[string]interface{}{
+				"crossplane.io/composition-resource-name": "resource-b",
+			},
+			"name": "test-render-b",
+		},
+		"spec": map[string]interface{}{
+			"coolerField": "I'm cooler!",
+		},
+	}
+)
+
+func TestFileLoader_Load(t *testing.T) {
```

Comment:
i think i saw nic say somewhere today to avoid using underscores in test names - there's a couple in this PR 😇